### PR TITLE
test(coq): duplicate theory in a workspace

### PR DESCRIPTION
**Pull request by Alizter**
Appears to be bugged since Dune does not complain at all.

Signed-off-by: Ali Caglayan <alizter@gmail.com>

### CODE REVIEW

File: test/blackbox-tests/test-cases/coq/duplicate-theory-workspace.t/run.t [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  > (coq.theory
+  >  (name foo))
+  > EOF
+
+  $ cat > B/dune << EOF
+  > (coq.theory
+  >  (name foo))
+  > EOF
+
+BUG this is not caught
```
Comment (ejgallego) [inline test/blackbox-tests/test-cases/coq/duplicate-theory-workspace.t/run.t:13]: I guess we are lazier now, as the theories are not public, `dune build @install` won't complain.

Comment (Alizter) [reply test/blackbox-tests/test-cases/coq/duplicate-theory-workspace.t/run.t:13]: Oh yes let me check again.

Comment (Alizter) [reply test/blackbox-tests/test-cases/coq/duplicate-theory-workspace.t/run.t:13]: As you can see now, they are both being built.

Comment (ejgallego) [reply test/blackbox-tests/test-cases/coq/duplicate-theory-workspace.t/run.t:13]: I guess that's OK as they are private theories, right?

Comment (Alizter) [reply test/blackbox-tests/test-cases/coq/duplicate-theory-workspace.t/run.t:13]: Oh yes, I guess the package layer restricts this from being an issue.

Comment (ejgallego) [reply test/blackbox-tests/test-cases/coq/duplicate-theory-workspace.t/run.t:13]: Yes, they will never make it to the workspace theory db as they are not public.

**Pull request closed without merging**